synapse/app/generic_worker.py

```diff
@@ -961,6 +962,22 @@ def start(config_options):
         # For other worker types we force this to off.
         config.worker.send_federation = False
 
+    if config.worker_app == "synapse.app.background_worker":
+        if config.worker.run_background_tasks:
+            sys.stderr.write(
+                "\nThe run_background_tasks must be disabled in the main synapse process"
+                "\nbefore they can be run in a separate worker."
+                "\nPlease add ``run_background_tasks: false`` to the main config"
+                "\n"
+            )
+            sys.exit(1)
+
+            # Force the background tasks to start since they will be disabled in the main config
+            config.worker.run_background_tasks = True
+        else:
+            # For other worker types we force this to off.
+            config.worker.run_background_tasks = False
```

Comment:
Ideally I'd prefer we added some config such as:

```yaml
run_background_task_on: "some_instance"
```

as I kinda want to remove the different worker apps and just have a single one that is controlled purely via config.  c.f. `media_instance_running_background_jobs` config option.

Reply:
That makes sense. I've made the requested change and removed this check.